src/NZSL/GlslWriter.cpp

```diff
@@ -272,6 +272,18 @@ namespace nzsl
 			void Visit(Ast::DeclareStructStatement& node) override
 			{
 				structs[node.structIndex.value()] = &node.description;
+				if (node.description.layout.HasValue())
+				{
+					const auto &layout = node.description.layout.GetResultingValue();
+					switch (layout) {
+						case Ast::MemoryLayout::Std430:
+						case Ast::MemoryLayout::Scalar:
+							// GL_EXT_scalar_block_layout (required for layout(scalar) and layout(std430))
+							requiredExtensions.emplace("GL_EXT_scalar_block_layout");
```

Comment:
from the extension, GL_EXT_scalar_block_layout is only required for layout std430 on uniform blocks but not on storage blocks

Reply:
Hmm. Should I just leave the case for Std430 then? Or can you point out a way for me to check for struct types. Also I moved these stuff to `Resolve()` function.

Reply:
Struct are not tied to uniform/storage in NZSL, you can declare a struct and use it as a SSBO and as a uniform block without redeclaring it.

So what we need to detect is a storage type like `storage[myStruct]`, it's quite easy to do it by detecting uniform external variables (https://github.com/NazaraEngine/ShaderLang/blob/main/src/NZSL/GlslWriter.cpp#L188) and retrieving the previously registered struct (https://github.com/NazaraEngine/ShaderLang/blob/main/src/NZSL/GlslWriter.cpp#L310) and checking its memory layout at this point.